session.go

```diff
@@ -286,7 +291,12 @@ func (s *Session) keepalive() {
 	for {
 		select {
 		case <-time.After(s.config.KeepAliveInterval):
-			s.Ping()
+			_, err := s.Ping()
+			if err != nil {
+				s.logger.Printf("[ERR] yamux: keepalive failed: %v", err)
+				s.exitErr(ErrKeepAliveTimeout)
+				return
+			}
```

Comment:
You mentioned checking the `Close()` error, but I'm not sure what I would do with it. I switched to using your usual pattern of `exitErr()`.